### PR TITLE
Add value objects, domain errors and migration for Sprints

### PR DESCRIPTION
**Pull request by pzajaczkowski**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205364838417508

### CODE REVIEW

File: src/core/Codend.Persistence/Configurations/SprintConfiguration.cs [hunk 1]
```diff
@@ -14,6 +15,31 @@ internal sealed class SprintConfiguration : IEntityTypeConfiguration<Sprint>
...
         builder.ConfigureKeyId((Guid guid) => new SprintId(guid));
+        
+        builder
+            .OwnsOne(sprint => sprint.Period,
+            sprintBuilder =>
+            {
+                sprintBuilder
+                    .Property(period => period.StartDate)
+                    .HasColumnName(nameof(SprintPeriod.StartDate))
+                    .IsRequired();
```
Comment (maksimowiczm) [inline src/core/Codend.Persistence/Configurations/SprintConfiguration.cs:26]: Specify DateTime precision

File: src/core/Codend.Persistence/Configurations/SprintConfiguration.cs [hunk 2]
```diff
@@ -14,6 +15,31 @@ internal sealed class SprintConfiguration : IEntityTypeConfiguration<Sprint>
...
+                
+                sprintBuilder
+                    .Property(period => period.EndDate)
+                    .HasColumnName(nameof(SprintPeriod.EndDate))
+                    .IsRequired();
```
Comment (maksimowiczm) [inline src/core/Codend.Persistence/Configurations/SprintConfiguration.cs:31]: Specify DateTime precision

**Review by maksimowiczm — CHANGES_REQUESTED** [review]

File: src/core/Codend.Domain/ValueObjects/SprintPeriod.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    /// Creates <see cref="SprintPeriod"/> instance.
+    /// </summary>
+    /// <param name="startDate">StartDate value.</param>
+    /// <param name="endDate">EndDate value.</param>
+    /// <returns>The result of sprint period creation. Returns sprint period or an error.</returns>
+    public static Result<SprintPeriod> Create(DateTime startDate, DateTime endDate)
+    {
+        return Result
+            .Ok(new SprintPeriod(startDate, endDate))
+            .Ensure(() => startDate.CompareTo(endDate) < 0, new DomainErrors.ProjectVersionTag.TagTooLong());
```
Comment (maksimowiczm) [inline src/core/Codend.Domain/ValueObjects/SprintPeriod.cs:40]: ☠☠☠☠☠☠☠☠☠☠☠☠☠

**Review by maksimowiczm — CHANGES_REQUESTED** [review]

File: src/core/Codend.Domain/ValueObjects/SprintPeriod.cs (hunk 3 above)
Comment (pzajaczkowski) [reply src/core/Codend.Domain/ValueObjects/SprintPeriod.cs:40]: xD

**Review by maksimowiczm — APPROVED** [review]

**Pull request merged**